Add ability to customize left/right Y axis labels

kennethnym opened the pull request:
## Description
This PR adds a textbox in the Y axis config panel that allows the user to change the left or right axis label, whichever is selected currently.

## Testing instructions
- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
connect to DSEGOG-197

kennethnym committed:
Add ability to customize y axis labels

kennethnym committed:
Write unit tests for the new changes

kennethnym committed:
Add e2e test for the new changes

kennethnym committed:
Merge branch 'develop' into DSEGOG-197-customize-axis-label

kennethnym committed:
Try to fix playwright tests

louise-davies commented:
@kennethnym can you take a look at the failing tests on this PR? (not urgent, just probably do this before tackling any new issues) I think this was the one which had some broken packages on the branch, so merging in develop might help fix things

kennethnym committed:
Update test snapshots

kennethnym committed:
Merge branch 'develop' into DSEGOG-197-customize-axis-label

# Conflicts:
#	src/plotting/__snapshots__/plot.component.test.tsx.snap

kennethnym committed:
Merge branch 'develop' into DSEGOG-197-customize-axis-label

kennethnym committed:
Add test snapshots for axis label customization

louise-davies committed:
DSEGOG-197 add firefox e2e snapshot for Y axis label customisation

kennethnym committed:
Merge branch 'develop' into DSEGOG-197-customize-axis-label

kennethnym committed:
Merge branch 'develop' into DSEGOG-197-customize-axis-label

kennethnym committed:
Update playwright webkit test snapshot

louise-davies commented on src/plotting/plotSettings/yAxisTab.component.tsx:
```diff
@@ -231,6 +238,23 @@ const YAxisTab = (props: YAxisTabProps) => {
     ]
   );
 
+  const changeAxisLabel = React.useCallback(
```
There's no unit test covering this new function - hence why the patch coverage is failing

kennethnym committed:
Merge remote-tracking branch 'origin/DSEGOG-197-customize-axis-label' into DSEGOG-197-customize-axis-label

kennethnym committed:
Add unit test for testing axis label customization

kennethnym committed:
Remove redundant jest.fn mock

Remove jest.fn mock used in plotSettingsTextField test. The onChange function is not inspected, so there is no point in wrapping the state setter in a jest fn.

louise-davies commented on e2e/plotting.spec.ts:
```diff
@@ -527,3 +527,34 @@ test('user can plot channels on the right y axis', async ({ page }) => {
+
+  await popup.getByRole('option', { name: 'Shot Number', exact: true }).click();
+  await popup.getByRole('textbox', { name: 'Label' }).type('left y axis');
+  await popup.getByRole('button', { name: 'Right' }).click();
+  await popup.getByRole('textbox', { name: 'Label' }).type('right y axis');
```
This test adds a right y axis label, but never selects a channel to put on the right Y axis for it to show up. Either add a channel & update the snapshots to test the right Y axis label (probably better to test it), or remove this code.

louise-davies reviewed (changes requested):
Saving the plot doesn't save the axis labels - looking at the code `handleSavePlot` hasn't been changed, so presumably it's not passing the axis labels to be saved

kennethnym committed:
Pass x/y-axis labels to config object when saving

kennethnym committed:
Merge branch 'develop' into DSEGOG-197-customize-axis-label

kennethnym committed:
Improve testing for customizing axis labels

louise-davies commented on src/plotting/plotWindow.component.tsx:
```diff
@@ -70,6 +71,8 @@ const PlotWindow = (props: PlotWindowProps) => {
   const [rightYAxisScale, setRightYAxisScale] = React.useState<YAxisScale>(
     plotConfig.rightYAxisScale
   );
+  const [leftYAxisLabel, setLeftYAxisLabel] = React.useState('');
+  const [rightYAxisLabel, setRightYAxisLabel] = React.useState('');
```
These aren't initialising from the `plotConfig`, so even though the axis labels are saving to Redux, when you reopen a plot after saving the labels aren't showing up.

kennethnym committed:
Load initial axis labels from saved plot config

louise-davies reviewed (approved):
LGTM
